prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -0,0 +1,341 @@
+import com.microsoft.azure.functions.annotation.TimerTrigger
+import com.microsoft.azure.functions.annotation.StorageAccount
+
+import gov.cdc.prime.router.azure.db.tables.pojos.ReportFile
+import gov.cdc.prime.router.azure.WorkflowEngine
+
+import java.util.Date;
+import java.time.ZoneOffset;
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.fasterxml.jackson.annotation.JsonProperty
+
+import java.time.ZonedDateTime
+import java.time.Duration
+import java.time.format.DateTimeFormatter
+import com.cronutils.model.time.ExecutionTime;
+import com.cronutils.parser.CronParser;
+import com.cronutils.model.definition.CronDefinitionBuilder;
+import com.cronutils.model.CronType;
+
+import com.sendgrid.helpers.mail.objects.*;
+import com.sendgrid.*;
+import com.sendgrid.helpers.mail.Mail;
+
+import gov.cdc.prime.router.azure.db.enums.SettingType;
+import org.jooq.Configuration
+
+import java.io.IOException;
+
+import khttp.get as httpGet
+import org.json.JSONObject
+import org.json.JSONArray
+
+import gov.cdc.prime.router.secrets.SecretHelper;
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+
+import com.microsoft.azure.functions.HttpMethod
+import com.microsoft.azure.functions.HttpRequestMessage
+import com.microsoft.azure.functions.HttpResponseMessage
+import com.microsoft.azure.functions.HttpStatus
+import com.microsoft.azure.functions.annotation.AuthorizationLevel
+import com.microsoft.azure.functions.annotation.BindingName
+import com.microsoft.azure.functions.annotation.HttpTrigger
+
+import com.okta.jwt.JwtVerifiers
+import java.util.logging.Level
+import java.util.logging.Logger
+
+const val SCHEDULE = "*/5 * * * *" // every 5 minutes
+const val OKTA_ISSUER = "https://hhs-prime.okta.com/oauth2/default"
+const val OKTA_GROUPS_API = "https://hhs-prime-admin.okta.com/api/v1/groups"
+const val FROM_EMAIL = "reportstream@cdc.gov"
+const val SUBJECT_EMAIL = "ReportStream Daily Email"
+
+
+
+data class EmailSchedule ( 
+    val template: String,
+    val type: String,
+    val cronSchedule: String,
+    val organizations: List<String> = ArrayList<String>(),
+    val emails: List<String> = ArrayList<String>(),
+    val parameters: Map<String,String> = HashMap<String,String>()
+) {}
+ 
+class EmailScheduleEngine  {
+
+    val workflowEngine = WorkflowEngine()
+
+    /**
+     * Create Email Schedule
+     */
+    @FunctionName("createEmailSchedule")
+    @StorageAccount("AzureWebJobsStorage")
+    fun createEmailSchedule(
+        @HttpTrigger(
+            name = "createEmailSchedule",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "email-schedule"
+        ) request: HttpRequestMessage<String?>,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        var user:String? = validateUser( request, context.logger );
```

Comment:
Reading `validateUser`, any user with a valid okta jwt token containing a subject (`sub`) will be able to create (and below: delete) e-mail schedules. Is this desired? Does there need to be any validation that you are touching only those schedules you are allowed to touch?

Reply:
Good point - should be active for only DHPRIMEADMINS; changing